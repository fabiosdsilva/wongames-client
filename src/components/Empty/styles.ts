import styled from 'styled-components'

export const Wrapper = styled.main``

export const Image = styled.img``

export const Title = styled.h3``

export const Description = styled.p``
