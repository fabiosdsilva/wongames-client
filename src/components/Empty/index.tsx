import Button from 'components/Button'
import Link from 'next/link'
import * as S from './styles'

export type EmptyProps = {
  title: string
  description: string
  hasLink?: boolean
}

const Empty = ({ title, description, hasLink }: EmptyProps) => (
  <S.Wrapper>
    <S.Image
      src="img/empty.svg"
      aria-label="A gamer in couch playing videogamer"
      role="img"
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>

    {hasLink && (
      <Link href="/" passHref>
        <Button as="a">Go to back home</Button>
      </Link>
    )}
  </S.Wrapper>
)

export default Empty
