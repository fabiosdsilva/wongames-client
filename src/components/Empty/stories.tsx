import { Story, Meta } from '@storybook/react'
import Empty, { EmptyProps } from '.'

export default {
  title: 'Empty',
  component: Empty,
  parameters: {
    backgrounds: {
      default: 'won-dark'
    }
  },
  args: {
    title: 'A simple title',
    description: 'a simple description',
    hasLink: true
  }
} as Meta

export const Default: Story<EmptyProps> = (args) => <Empty {...args} />
