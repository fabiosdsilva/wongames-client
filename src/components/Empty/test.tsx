import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Empty, { EmptyProps } from '.'

const props: EmptyProps = {
  title: 'A simple title',
  description: 'A simple description'
}

describe('<Empty />', () => {
  it('verificar se está renderizando a imagem, title e descrição', () => {
    renderWithTheme(<Empty {...props} />)

    expect(
      screen.getByRole('img', { name: /a gamer in couch playing videogamer/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /a simple title/i })
    ).toBeInTheDocument()
    expect(screen.getByText(/a simple description/i)).toBeInTheDocument()
  })

  it('verificar se está o link', () => {
    renderWithTheme(<Empty {...props} hasLink />)

    expect(
      screen.getByRole('link', { name: /go to back home/i })
    ).toHaveAttribute('href', '/')
  })

  it('verificar se está o link está destivado', () => {
    renderWithTheme(<Empty {...props} />)

    expect(
      screen.queryByRole('link', { name: /go to back home/i })
    ).not.toBeInTheDocument()
  })
})
