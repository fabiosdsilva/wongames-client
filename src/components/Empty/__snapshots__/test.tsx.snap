// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<Empty /> verificar se está renderizando a imagem, title e descrição 1`
] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c1 {
  max-width: 100%;
}

.c2 {
  color: #F231A5;
  font-size: 2.8rem;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

<body>
  <div>
    <main
      class="c0"
    >
      <img
        aria-label="A gamer in couch playing videogamer"
        class="c1"
        role="img"
        src="img/empty.svg"
      />
      <h2
        class="c2"
      >
        A simple title
      </h2>
      <p
        class="c3"
      >
        A simple description
      </p>
    </main>
  </div>
</body>
`
